actions/actions.py

```diff
@@ -9,27 +9,46 @@
 
 from rasa_sdk import Action, Tracker
 from rasa_sdk.executor import CollectingDispatcher
-import requests
+import aiohttp
+import json
+import logging
 
+logger = logging.getLogger(__name__)
 
 class ActionHaystack(Action):
 
     def name(self) -> Text:
         return "call_haystack"
 
-    def run(self, dispatcher: CollectingDispatcher,
+    async def run(self, dispatcher: CollectingDispatcher,
             tracker: Tracker,
             domain: Dict[Text, Any]) -> List[Dict[Text, Any]]:
 
         url = "http://localhost:8000/query"
-        payload = {"query": str(tracker.latest_message["text"])}
+        # params={'filters': {}, 'Retriever': {'top_k': 3}, 'Reader': {'top_k': 3}, 'Query': {'debug': False}}
+        payload = {
+            "query": str(tracker.latest_message["text"]),
+            "params":{"filters": {}, "Retriever": {"top_k": 3}, "Reader": {"top_k": 3}, "Query": {"debug": False}}
+        }
         headers = {
             'Content-Type': 'application/json'
         }
-        response = requests.request("POST", url, headers=headers, json=payload).json()
+
+        logger.debug(f"Calling {url}, ask: {str(tracker.latest_message['text'])}")
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.post(url, headers=headers, json=payload) as response:
+                    response = response.json()
```

Comment:
we need an await here, don't we?
```suggestion
                    response = await response.json()
```